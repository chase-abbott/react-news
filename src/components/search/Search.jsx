import React, { useState } from 'react';

const Search = () => {
  const [inputValue, setInputValue] = useState('');

  return (
    <form>
      <label htmlFor="search-input"></label>
      <input 
        type="text" 
        id="search-input"
        value={inputValue}
        onChange={setInputValue}
      ></input>
      <button> Submit </button>
    </form>
  );
};


export default Search;
