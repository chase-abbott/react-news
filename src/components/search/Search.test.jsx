import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search.jsx';

describe('Search componenet', () => {
  it('renders a form element with input and search button', () => {
    const { asFragment } = render(<Search/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
