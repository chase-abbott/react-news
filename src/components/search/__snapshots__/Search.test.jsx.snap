// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search componenet renders a form element with input and search button 1`] = `
<DocumentFragment>
  <form>
    <label
      for="search-input"
    />
    <input
      id="search-input"
      type="text"
      value=""
    />
    <button>
       Submit 
    </button>
  </form>
</DocumentFragment>
`;
